File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/map/GraphicControllerSink.java
package com.valentingrigorean.arcgis_maps_flutter.map;

import android.graphics.Color;

import com.esri.arcgisruntime.geometry.Geometry;
import com.esri.arcgisruntime.mapping.view.GraphicsOverlay;

public interface GraphicControllerSink {
    void add(GraphicsOverlay graphicsOverlay);

    void remove(GraphicsOverlay graphicsOverlay);

    boolean getVisible();

    void setVisible(boolean visible);

    void setZIndex(int zIndex);

    void setGeometry(Geometry geometry);

    boolean isSelected();

    void setSelected(boolean selected);

    boolean canConsumeTapEvents();

    void setConsumeTapEvents(boolean consumeTapEvent);

    void setSelectedColor(Color color);
}

File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/map/GraphicControllerSink.kt
package com.valentingrigorean.arcgis_maps_flutter.map

import android.graphics.Color
import com.arcgismaps.geometry.Geometry
import com.arcgismaps.mapping.view.GraphicsOverlay

interface GraphicControllerSink {
    fun add(graphicsOverlay: GraphicsOverlay)
    fun remove(graphicsOverlay: GraphicsOverlay)
    var visible: Boolean
    fun setZIndex(zIndex: Int)
    fun setGeometry(geometry: Geometry?)

    var isSelected: Boolean
    fun canConsumeTapEvents(): Boolean
    fun setConsumeTapEvents(consumeTapEvent: Boolean)
    fun setSelectedColor(color: Color?)
}